tests/test_io.py

```diff
@@ -81,3 +89,209 @@ def test_read_sofa_position_type_unit(generate_sofa_postype_error):
+    """ Check if a TypeError is raised when writing an arbitrary
+    object.
+    """
+    filename = os.path.join(tmpdir, 'test_anyObj.far')
+    with pytest.raises(TypeError):
+        io.write(filename, any_obj=any_obj)
+
+
+def test_read_orientations(orientations, tmpdir):
+    """ Orientations
+    Make sure `read` understands the bits written by `write`
+    """
+    filename = os.path.join(tmpdir, 'orientations.far')
+    io.write(filename, orientations=orientations)
+    actual = io.read(filename)['orientations']
+    assert isinstance(actual, Orientations)
+    assert actual == orientations
+
+
+def test_read_coordinates(coordinates, tmpdir):
+    """ Coordinates
+    Make sure `read` understands the bits written by `write`
+    """
+    filename = os.path.join(tmpdir, 'coordinates.far')
+    io.write(filename, coordinates=coordinates)
+    actual = io.read(filename)['coordinates']
+    assert isinstance(actual, Coordinates)
+    assert actual == coordinates
+
+
+def test_read_signal(signal, tmpdir):
+    """ Signal
+    Make sure `read` understands the bits written by `write`
+    """
+    filename = os.path.join(tmpdir, 'signal.far')
+    io.write(filename, signal=signal)
+    actual = io.read(filename)['signal']
+    assert isinstance(actual, Signal)
+    assert actual == signal
+
+
+def test_read_sphericalvoronoi(sphericalvoronoi, tmpdir):
+    """ SphericalVoronoi
+    Make sure `read` understands the bits written by `write`
+    """
+    filename = os.path.join(tmpdir, 'sphericalvoronoi.far')
+    io.write(filename, sphericalvoronoi=sphericalvoronoi)
+    actual = io.read(filename)['sphericalvoronoi']
+    assert isinstance(actual, SphericalVoronoi)
+    assert actual == sphericalvoronoi
+
+
+def test_read_filter(filter, tmpdir):
+    """ Filter
+    Make sure `read` understands the bits written by `write`
+    """
+    filename = os.path.join(tmpdir, 'filter.far')
+    io.write(filename, filter=filter)
+    actual = io.read(filename)['filter']
+    assert isinstance(actual, fo.Filter)
+    assert actual == filter
+
+
+def test_write_filterIIR_TypeError(filterIIR, tmpdir):
+    """ FilterIIR
+    Can't be written to disk because objects store user-defined function.
+    """
+    filename = os.path.join(tmpdir, 'filterIIR.far')
+    with pytest.raises(TypeError):
+        io.write(filename, filterIIR=filterIIR)
+
+
+def test_write_filterFIR_TypeError(filterFIR, tmpdir):
+    """ FilterFIR
+    Can't be written to disk because objects store user-defined function.
+    """
+    filename = os.path.join(tmpdir, 'filterIIR.far')
+    with pytest.raises(TypeError):
+        io.write(filename, filterFIR=filterFIR)
+
+
+def test_write_filterSOS_TypeError(filterSOS, tmpdir):
+    """ FilterIIR
+    Can't be written to disk because objects store user-defined function.
+    """
+    filename = os.path.join(tmpdir, 'filterSOS.far')
+    with pytest.raises(TypeError):
+        io.write(filename, filterSOS=filterSOS)
```

Comment:
Can these tests be reformulated in a way, that they do not need to be deleted, if writing the filters is possible in the future?

Reply:
The tests here are representing the state-of-the-art requirement that there is not an implementation for those ojects right now on purpose. I think we should delete these tests when we open a new feature-branch for the implementation, I will  mention this in the docstring.